src/tsc-errors/classify-tsc-error.ts

```diff
@@ -0,0 +1,25 @@
+import { FilePathMatcher } from '../file-path-matcher';
+import { TscError, TscErrorMatchFlags } from './types';
+
+const classifyTscError = (
+  tscError: TscError,
+  ignoredErrorCodesSet: ReadonlySet<string>,
+  filePathMatcher: FilePathMatcher,
+): TscErrorMatchFlags => {
+  const errorCodeMatch = ignoredErrorCodesSet.has(tscError.tscErrorCode);
+  const filePathMatch = filePathMatcher.matches(tscError.filePath);
+
+  let matchFlags = TscErrorMatchFlags.filePath & TscErrorMatchFlags.errorCode;
```

Comment:
Very clever way of using bit flags instead of having 4 separate buckets like I had before :tada: 

Minor:

Could you set this to `0` explicitly? I had to think a bit why this is initialized the way it is, and then we use bitwise or. Only after a moment it occurred to me that this is simply a verbose way of initializing this to `0`

```suggestion
  let matchFlags = 0;
```

Reply:
I agree that it's a bit cryptic. I considered a `none = 0` flag but ultimately leaned against it because it was more of a pseudo-flag than an actual flag. I can resurrect that or find a better way to indicate the absence of any flags. Setting it to `0` explicitly works for me as well.

Reply:
IMO a `none = 0` case works well. Alternatively, add a `const noMatchFlags = 0;` next to the enum definition and use that as the initial value here. I reckon that would be less surprising and would also link the 2 parts of code together